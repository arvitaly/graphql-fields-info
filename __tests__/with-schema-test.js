"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const __1 = require("./..");
const schema_1 = require("./../__fixtures__/schema");
describe("Info with schema", () => {
    it("node query", () => {
        const query = `query Q1{
            node(id:"1"){
                ...F1
            }
        }
        fragment F1 on Model1{
            field1
        }
        `;
        const fieldsInfo = __1.fromQuery(query, schema_1.default);
        expect(fieldsInfo.getFields()).toMatchSnapshot();
    });
    it("connection with fragment query", () => {
        const query = `query Q1{
            viewer{
                model1(first:10){
                    pageInfo{
                        hasNextPage
                    }
                    ...F1
                }}
            }
            fragment F1 on Model1Connection{
                 edges{
                     node{
                         ...F2
                     }
                 }
            }
            fragment F2 on Model1{
                field1
                model2{
                    field2
                }
            }
        `;
        const fieldsInfo = __1.fromQuery(query, schema_1.default);
        expect(fieldsInfo.getFields()).toMatchSnapshot();
    });
});
