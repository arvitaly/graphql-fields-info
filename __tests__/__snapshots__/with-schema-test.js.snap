// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info with schema node query 1`] = `
Array [
  Object {
    "args": Array [
      Object {
        "defaultValue": undefined,
        "description": "The ID of an object",
        "name": "id",
        "type": "ID!",
      },
    ],
    "fields": Array [
      Object {
        "args": Array [],
        "fields": Array [],
        "isConnection": false,
        "isNode": false,
        "name": "field1",
        "node": Object {
          "alias": null,
          "arguments": Array [],
          "directives": Array [],
          "kind": "Field",
          "loc": Object {
            "end": 131,
            "start": 125,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 131,
              "start": 125,
            },
            "value": "field1",
          },
          "selectionSet": null,
        },
        "type": "String",
        "typeName": "Model1",
      },
    ],
    "isConnection": false,
    "isNode": false,
    "name": "node",
    "node": Object {
      "alias": null,
      "arguments": Array [
        Object {
          "kind": "Argument",
          "loc": Object {
            "end": 33,
            "start": 27,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 29,
              "start": 27,
            },
            "value": "id",
          },
          "value": Object {
            "kind": "StringValue",
            "loc": Object {
              "end": 33,
              "start": 30,
            },
            "value": "1",
          },
        },
      ],
      "directives": Array [],
      "kind": "Field",
      "loc": Object {
        "end": 71,
        "start": 22,
      },
      "name": Object {
        "kind": "Name",
        "loc": Object {
          "end": 26,
          "start": 22,
        },
        "value": "node",
      },
      "selectionSet": Object {
        "kind": "SelectionSet",
        "loc": Object {
          "end": 71,
          "start": 34,
        },
        "selections": Array [
          Object {
            "directives": Array [],
            "kind": "FragmentSpread",
            "loc": Object {
              "end": 57,
              "start": 52,
            },
            "name": Object {
              "kind": "Name",
              "loc": Object {
                "end": 57,
                "start": 55,
              },
              "value": "F1",
            },
          },
        ],
      },
    },
    "type": "Node",
  },
]
`;

exports[`Info with schema one query 1`] = `
Array [
  Object {
    "args": Array [],
    "fields": Array [
      Object {
        "args": Array [],
        "fields": Array [
          Object {
            "args": Array [],
            "fields": Array [
              Object {
                "args": Array [],
                "fields": Array [],
                "isConnection": false,
                "isNode": false,
                "name": "hasNextPage",
                "node": Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "loc": Object {
                    "end": 129,
                    "start": 118,
                  },
                  "name": Object {
                    "kind": "Name",
                    "loc": Object {
                      "end": 129,
                      "start": 118,
                    },
                    "value": "hasNextPage",
                  },
                  "selectionSet": null,
                },
                "type": "Boolean",
              },
            ],
            "isConnection": false,
            "isNode": false,
            "name": "pageInfo",
            "node": Object {
              "alias": null,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 151,
                "start": 84,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 92,
                  "start": 84,
                },
                "value": "pageInfo",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "loc": Object {
                  "end": 151,
                  "start": 92,
                },
                "selections": Array [
                  Object {
                    "alias": null,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "loc": Object {
                      "end": 129,
                      "start": 118,
                    },
                    "name": Object {
                      "kind": "Name",
                      "loc": Object {
                        "end": 129,
                        "start": 118,
                      },
                      "value": "hasNextPage",
                    },
                    "selectionSet": null,
                  },
                ],
              },
            },
            "type": "Model1ConnectionPageInfo",
          },
          Object {
            "args": Array [],
            "fields": Array [
              Object {
                "args": Array [],
                "fields": Array [
                  Object {
                    "args": Array [],
                    "fields": Array [],
                    "isConnection": false,
                    "isNode": false,
                    "name": "field1",
                    "node": Object {
                      "alias": null,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "loc": Object {
                        "end": 451,
                        "start": 445,
                      },
                      "name": Object {
                        "kind": "Name",
                        "loc": Object {
                          "end": 451,
                          "start": 445,
                        },
                        "value": "field1",
                      },
                      "selectionSet": null,
                    },
                    "type": "String",
                    "typeName": "Model1",
                  },
                  Object {
                    "args": Array [],
                    "fields": Array [
                      Object {
                        "args": Array [],
                        "fields": Array [],
                        "isConnection": false,
                        "isNode": false,
                        "name": "field2",
                        "node": Object {
                          "alias": null,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "loc": Object {
                            "end": 502,
                            "start": 496,
                          },
                          "name": Object {
                            "kind": "Name",
                            "loc": Object {
                              "end": 502,
                              "start": 496,
                            },
                            "value": "field2",
                          },
                          "selectionSet": null,
                        },
                        "type": "Int",
                      },
                    ],
                    "isConnection": false,
                    "isNode": true,
                    "name": "model2",
                    "node": Object {
                      "alias": null,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "loc": Object {
                        "end": 520,
                        "start": 468,
                      },
                      "name": Object {
                        "kind": "Name",
                        "loc": Object {
                          "end": 474,
                          "start": 468,
                        },
                        "value": "model2",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "loc": Object {
                          "end": 520,
                          "start": 474,
                        },
                        "selections": Array [
                          Object {
                            "alias": null,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "loc": Object {
                              "end": 502,
                              "start": 496,
                            },
                            "name": Object {
                              "kind": "Name",
                              "loc": Object {
                                "end": 502,
                                "start": 496,
                              },
                              "value": "field2",
                            },
                            "selectionSet": null,
                          },
                        ],
                      },
                    },
                    "type": "Model2",
                    "typeName": "Model1",
                  },
                ],
                "isConnection": false,
                "isNode": true,
                "name": "node",
                "node": Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "loc": Object {
                    "end": 360,
                    "start": 301,
                  },
                  "name": Object {
                    "kind": "Name",
                    "loc": Object {
                      "end": 305,
                      "start": 301,
                    },
                    "value": "node",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "loc": Object {
                      "end": 360,
                      "start": 305,
                    },
                    "selections": Array [
                      Object {
                        "directives": Array [],
                        "kind": "FragmentSpread",
                        "loc": Object {
                          "end": 337,
                          "start": 332,
                        },
                        "name": Object {
                          "kind": "Name",
                          "loc": Object {
                            "end": 337,
                            "start": 335,
                          },
                          "value": "F2",
                        },
                      },
                    ],
                  },
                },
                "type": "Model1",
              },
            ],
            "isConnection": false,
            "isNode": false,
            "name": "edges",
            "node": Object {
              "alias": null,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 379,
                "start": 273,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 278,
                  "start": 273,
                },
                "value": "edges",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "loc": Object {
                  "end": 379,
                  "start": 278,
                },
                "selections": Array [
                  Object {
                    "alias": null,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "loc": Object {
                      "end": 360,
                      "start": 301,
                    },
                    "name": Object {
                      "kind": "Name",
                      "loc": Object {
                        "end": 305,
                        "start": 301,
                      },
                      "value": "node",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "loc": Object {
                        "end": 360,
                        "start": 305,
                      },
                      "selections": Array [
                        Object {
                          "directives": Array [],
                          "kind": "FragmentSpread",
                          "loc": Object {
                            "end": 337,
                            "start": 332,
                          },
                          "name": Object {
                            "kind": "Name",
                            "loc": Object {
                              "end": 337,
                              "start": 335,
                            },
                            "value": "F2",
                          },
                        },
                      ],
                    },
                  },
                ],
              },
            },
            "type": "[Model1ConnectionEdge]!",
            "typeName": "Model1Connection",
          },
        ],
        "isConnection": true,
        "isNode": false,
        "name": "model1",
        "node": Object {
          "alias": null,
          "arguments": Array [
            Object {
              "kind": "Argument",
              "loc": Object {
                "end": 61,
                "start": 53,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 58,
                  "start": 53,
                },
                "value": "first",
              },
              "value": Object {
                "kind": "IntValue",
                "loc": Object {
                  "end": 61,
                  "start": 59,
                },
                "value": "10",
              },
            },
          ],
          "directives": Array [],
          "kind": "Field",
          "loc": Object {
            "end": 195,
            "start": 46,
          },
          "name": Object {
            "kind": "Name",
            "loc": Object {
              "end": 52,
              "start": 46,
            },
            "value": "model1",
          },
          "selectionSet": Object {
            "kind": "SelectionSet",
            "loc": Object {
              "end": 195,
              "start": 62,
            },
            "selections": Array [
              Object {
                "alias": null,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "loc": Object {
                  "end": 151,
                  "start": 84,
                },
                "name": Object {
                  "kind": "Name",
                  "loc": Object {
                    "end": 92,
                    "start": 84,
                  },
                  "value": "pageInfo",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "loc": Object {
                    "end": 151,
                    "start": 92,
                  },
                  "selections": Array [
                    Object {
                      "alias": null,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "loc": Object {
                        "end": 129,
                        "start": 118,
                      },
                      "name": Object {
                        "kind": "Name",
                        "loc": Object {
                          "end": 129,
                          "start": 118,
                        },
                        "value": "hasNextPage",
                      },
                      "selectionSet": null,
                    },
                  ],
                },
              },
              Object {
                "directives": Array [],
                "kind": "FragmentSpread",
                "loc": Object {
                  "end": 177,
                  "start": 172,
                },
                "name": Object {
                  "kind": "Name",
                  "loc": Object {
                    "end": 177,
                    "start": 175,
                  },
                  "value": "F1",
                },
              },
            ],
          },
        },
        "type": "Model1Connection",
      },
    ],
    "isConnection": false,
    "isNode": true,
    "name": "viewer",
    "node": Object {
      "alias": null,
      "arguments": Array [],
      "directives": Array [],
      "kind": "Field",
      "loc": Object {
        "end": 196,
        "start": 22,
      },
      "name": Object {
        "kind": "Name",
        "loc": Object {
          "end": 28,
          "start": 22,
        },
        "value": "viewer",
      },
      "selectionSet": Object {
        "kind": "SelectionSet",
        "loc": Object {
          "end": 196,
          "start": 28,
        },
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "loc": Object {
                  "end": 61,
                  "start": 53,
                },
                "name": Object {
                  "kind": "Name",
                  "loc": Object {
                    "end": 58,
                    "start": 53,
                  },
                  "value": "first",
                },
                "value": Object {
                  "kind": "IntValue",
                  "loc": Object {
                    "end": 61,
                    "start": 59,
                  },
                  "value": "10",
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "loc": Object {
              "end": 195,
              "start": 46,
            },
            "name": Object {
              "kind": "Name",
              "loc": Object {
                "end": 52,
                "start": 46,
              },
              "value": "model1",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "loc": Object {
                "end": 195,
                "start": 62,
              },
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "loc": Object {
                    "end": 151,
                    "start": 84,
                  },
                  "name": Object {
                    "kind": "Name",
                    "loc": Object {
                      "end": 92,
                      "start": 84,
                    },
                    "value": "pageInfo",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "loc": Object {
                      "end": 151,
                      "start": 92,
                    },
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "loc": Object {
                          "end": 129,
                          "start": 118,
                        },
                        "name": Object {
                          "kind": "Name",
                          "loc": Object {
                            "end": 129,
                            "start": 118,
                          },
                          "value": "hasNextPage",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
                Object {
                  "directives": Array [],
                  "kind": "FragmentSpread",
                  "loc": Object {
                    "end": 177,
                    "start": 172,
                  },
                  "name": Object {
                    "kind": "Name",
                    "loc": Object {
                      "end": 177,
                      "start": 175,
                    },
                    "value": "F1",
                  },
                },
              ],
            },
          },
        ],
      },
    },
    "type": "Viewer",
  },
]
`;
